### ISSUE TITLE
Error loading data source package 'Biber::Output::bibtex': Biber/Output/bibtex.pm did not return a true value at (eval 413) line 1.

### ISSUE THREAD

**Issue by ClaudiusL** [plk/biber#334]
I used to let biber generate a subset of my BIB file, containing only cited items. This worked flawlessly with biber 2.13 but fails after updating to biber 2.15 (beta). Any help is appreciated!

**Problem:**
- biber 2.15 (beta) (on Windows 10 using MiKTeX)
- command: `biber --output_file=out.bib --output_format=bibtex document`
- output:
```
INFO - This is Biber 2.15 (beta)
INFO - Logfile is 'document.blg'
ERROR - Error loading data source package 'Biber::Output::bibtex': Biber/Output/bibtex.pm did not return a true value at (eval 413) line 1.

INFO - ERRORS: 1
```
- expected output: with biber 2.13, the same command writes only the entries cited in "document.tex" to "out.bib".

**Reproducible example**:

```
%document.tex
\documentclass{scrartcl}

\begin{filecontents}{mybib.bib}
@Article{Test2020,
	author       = {Test, The},
	date         = {2020-09-08},
	journaltitle = {Journal of Tests},
	title        = {A Test Article},
}

@Article{Test2020b,
	author       = {Test, The},
	date         = {2020-09-08},
	journaltitle = {Journal of Tests},
	title        = {A Second Test},
}

\end{filecontents}

\usepackage{biblatex}
\addbibresource{mybib.bib}

\begin{document}
	\textcite{Test2020b}
\end{document}
```
```
% expected output: out.bib

@ARTICLE{Test2020b,
  AUTHOR = {Test, The},
  DATE = {2020-09-08},
  JOURNALTITLE = {Journal of Tests},
  TITLE = {A Second Test},
}
```

**Comment by plk** [comment]
Is there a reason you are using the beta version of 2.15? 2.15 is released now - can you try to update your installation and also run `biber --cache` and delete whichever folder this reports?

**Comment by moewew** [comment]
@plk The Biber installed by MikTeX (at least the 64bit version) identifies as `biber version: 2.15 (beta)`. This doesn't appear to be a MikTeX issue since the Biber from CTAN (https://www.ctan.org/tex-archive/biblio/biber/binaries/Windows) also says '(beta)'. The binary on SF (https://sourceforge.net/projects/biblatex-biber/files/biblatex-biber/current/binaries/Windows/) also has '(beta)'.

**Comment by ClaudiusL** [comment]
@plk As @moewew wrote, biber 2.15 (beta) is what MiKTeX installs. I installed a full version of MiKTeX yesterday and checked for updates today. Yet, the biber version is 2.15 (beta).

I tried deleting the directory reported by `biber --cache` but this had no effect.

**Comment by plk** [comment]
That is very strange - the other binaries appear to be non-beta flagged as they should be. Can you try the 2.15 from sourceforge and just overwrite your current 2.15 binary? That error looks like a corrupted install if the cache clearing didn't fix it.

**Comment by ClaudiusL** [comment]
I just downloaded biber from https://sourceforge.net/projects/biblatex-biber/files/biblatex-biber/current/binaries/Windows/biber-MSWIN64.zip/download . This version is also 2.15 (beta).

Did you mean another version?

**Comment by plk** [comment]
No, that's correct for now but I just wanted to see if this version actually works for you without the error.

**Comment by ClaudiusL** [comment]
No, unfortunately it doesn't. 
When I download 2.14 the error vanishes – but the (expected) error about the incompatible biblatex version occurs instead.

**Comment by plk** [comment]
Can you try getting 2.15 (current folder) from SF again now? I have rebuilt and re-uploaded there. Just make sure that the date on the version you get says something sensible to suggest it was updated in the last few minutes as SF takes a while to replicate round the world.

**Comment by ClaudiusL** [comment]
I downloaded the new version (the last modified date is about 19 minutes in the past), but it still self-identifies as 2.15 (beta) and still throws the same error.

**Comment by ClaudiusL** [comment]
To make sure that all this is not just due to something weird going on with my machine, I just took my spare laptop (with biber 2.13) and updated MiKTeX. Same issue.

**Comment by plk** [comment]
Ah, my build farm win10 VM has a corrupt git clone ... looking into it.

**Comment by ClaudiusL** [comment]
Thanks for your effort, plk!

**Comment by plk** [comment]
Please get 2.15 again from SF - should just have been updated - it seems my build server had a corrupted git clone and so was building an old, broken version ...

**Comment by ClaudiusL** [comment]
The new version solves the problem – thank you!

However, I noticed another issue that seems to be new in 2.15: As soon as the directory contains a (valid) `biber.conf` file (the actual content does not seem to matter), the following message is printed:

    Log4perl: Seems like no initialization happened. Forgot to call init()?

After this message biber apparently works fine.

**Comment by plk** [comment]
Hmm, I can't reproduce this with either 2.15 or 2.16 DEV - where do you see this, in the standard console output?

**Comment by ClaudiusL** [comment]
Here's a reproducible example: I'm using `mybib.bib` from the initial posting and the following `biber.conf`:

```
<?xml version="1.0" encoding="UTF-8" ?>
<config>
	<sourcemap>
		<maps datatype="bibtex" level="user">
			<map map_overwrite="1">
				<map_step map_field_set="author" map_field_value="Cool Guy" />
			</map>
		</maps>
	</sourcemap>
</config>
```

Then I run `biber --tool --output_file=test.bib mybib.bib` on the command line. Version 2.15 reports:
```

>biber --tool --output_file=test.bib mybib.bib
Log4perl: Seems like no initialization happened. Forgot to call init()?
INFO - This is Biber 2.15 running in TOOL mode
INFO - Config file is 'C:\Users\loehne01\TMP_lokal\Paper_CL\biber.conf'
INFO - Logfile is 'mybib.bib.blg'
INFO - Looking for bibtex format file 'mybib.bib'
INFO - LaTeX decoding ...
INFO - Found BibTeX data source 'mybib.bib'
INFO - Overriding locale 'en_US' defaults 'variable = shifted' with 'variable = non-ignorable'
INFO - Overriding locale 'en_US' defaults 'normalization = NFD' with 'normalization = prenormalized'
INFO - Sorting list 'tool/global//global/global' of type 'entry' with template 'tool' and locale 'en_US'
INFO - No sort tailoring available for locale 'en_US'
INFO - Writing 'test.bib' with encoding 'UTF-8'
INFO - Output to test.bib
```

`test.bib` is as expected. With 2.14 I get the same output, except for the first line with `Log4perl` (and two INFOs about "globbing data source").

I'm using an example with tool mode because 2.14 is incompatible with my `biblatex` version. However, if I just run biber 2.15 on a regular TEX document, I also get the `Log4perl` message (either on the command line after `biber document` or in TeXStudio's messages pane).

I would have tested with 2.16 DEV, but could not find it. The "development" version on SF is 2.15 (beta).

**Comment by plk** [comment]
Hmm, I still can't reproduce this with 2.15 or 2.16 DEV in tool or normal mode. I tried your `biber.conf` - also fine. The only thing I haven't tried is your `.bib` - perhaps you can post a minimal .bib that also shows this? I have updated the development `biber` folder with 2.16 beta to try too.

**Comment by ClaudiusL** [comment]
Thanks for investigating!

I'm using `mybib.bib` generated by the `filecontents` environment in the TEX document I postet initially.

2.16 beta shows the `Log4perl`  message just like 2.15.

Tomorrow I can test on a different machine – I'm curious to see if *I* can reproduce the issue.

**Comment by moewew** [comment]
I tested with the dev binaries for Windows and the `Log4perl` message appears for me as well. But only in the terminal output and not in the `.blg` file.

**Comment by plk** [comment]
Windows or linux?

**Comment by moewew** [comment]
I tested on Windows 10 (and I think so did @ClaudiusL if I read the first post correctly).

**Comment by ClaudiusL** [comment]
Sorry, could not make it yesterday.

On another machine (yes: also Windows) I get the same behavior:

- 2.14 works fine.
- 2.15 has the `Log4perl` message in the command line output (but not in the BLG file).
- 2.16 (beta) behaves as 2.15.

By the way, MiKTeX still serves the broken 2.15 (beta) – would it be possible to update this?

**Comment by plk** [comment]
I can reproduce this on Windows but not on OSX - @moewew - can you reproduce this on Linux?

**Comment by plk** [comment]
I see the issue, it's a harmless logging issue with the new code for dealing with UTF-8 filenames on windows. Should be fixed in 2.16 DEV now.

**Comment by ClaudiusL** [comment]
I can confirm that 2.16 DEV fixes the message on Windows.

Thanks, plk!
